packages/flagging/src/precomputeClient.ts

```diff
@@ -0,0 +1,183 @@
+        try {
+          const parsed = JSON.parse(value)
+          return typeof parsed === 'object' && parsed !== null ? parsed : defaultValue
+        } catch {
+          return defaultValue
+        }
+      }
+      return typeof value === 'object' && value !== null ? value : defaultValue
+    })
+  }
+
+  private getPrecomputedAssignment<T>(
+    flagKey: string,
+    defaultValue: T,
+    _expectedType: VariationType,
+    valueTransformer: (value: unknown) => T = (v) => v as T
+  ): T {
+    // validateNotBlank(flagKey, 'Invalid argument: flagKey cannot be blank')
+
+    const precomputedFlag = this.getPrecomputedFlag(flagKey)
+
+    if (precomputedFlag === null) {
+      return defaultValue
+    }
+
+    const result: FlagEvaluationWithoutDetails = {
+      flagKey,
+      format: 'PRECOMPUTED',
+      subjectKey: this.subject.key ?? '',
+      subjectAttributes: {
+        ...this.subject.attributes?.numericAttributes,
+        ...this.subject.attributes?.categoricalAttributes,
+      },
+      variation: {
+        key: precomputedFlag.variationKey ?? '',
+        value: precomputedFlag.variationValue,
+      },
+      allocationKey: precomputedFlag.allocationKey ?? '',
+      extraLogging: precomputedFlag.extraLogging ?? {},
+      doLog: precomputedFlag.doLog,
+      entityId: null,
+    }
+
+    try {
+      const transformedValue =
+        result.variation?.value !== undefined ? valueTransformer(result.variation.value) : defaultValue
+      // TODO: add logging or open feature hook
+      // if (result?.doLog) {
+      //   this.logAssignment(result)
+      // }
+      return transformedValue
+    } catch {
+      return defaultValue
+    }
+  }
+
+  private getPrecomputedFlag(flagKey: string): PrecomputedFlag | null {
+    return this.precomputedFlagStore.get(flagKey) ?? null
+  }
+}
+
+/**
+ * Initializes the precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - precomputed client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export function offlinePrecomputedInit(config: IPrecomputedClientConfigSync): PrecomputeClient | null {
+  let configurationWire: IConfigurationWire
+  try {
+    configurationWire = JSON.parse(config.precomputedConfiguration)
+    if (!configurationWire.precomputed) {
+      throw new Error('Invalid precomputed configuration wire: missing precomputed field')
+    }
+  } catch {
+    const errorMessage = 'Invalid precomputed configuration wire'
+    if (config.throwOnFailedInitialization) {
+      throw new Error(errorMessage)
+    }
+    return null
+  }
+
+  const { subjectKey, subjectAttributes, response } = configurationWire.precomputed
+  const parsedResponse: IPrecomputedConfigurationResponse = JSON.parse(response)
```

Comment:
should we have a try-catch here in case response is bonked?

Same for defensively checking `flags` exists.